fluss-server/src/main/java/com/alibaba/fluss/server/tablet/TabletService.java

```diff
@@ -312,7 +314,38 @@ public CompletableFuture<ListOffsetsResponse> listOffsets(ListOffsetsRequest req
 
     @Override
     public CompletableFuture<InitWriterResponse> initWriter(InitWriterRequest request) {
-        // todo: add authorization for table acl until https://github.com/alibaba/fluss/issues/756.
+        List<PbTablePath> tablePathsList = request.getTablePathsList();
+        if (authorizer != null) {
+            if (tablePathsList == null || tablePathsList.isEmpty()) {
+                throw new UnknownTableOrBucketException(
+                        "The table paths can not be null or empty when initWriter.");
+            }
+
+            boolean hasAuthorizedPath = false;
+            for (PbTablePath pbTablePath : tablePathsList) {
+                if (authorizer.isAuthorized(
+                        currentSession(),
+                        WRITE,
+                        Resource.table(
+                                pbTablePath.getDatabaseName(), pbTablePath.getTableName()))) {
+                    hasAuthorizedPath = true;
+                    break;
```

Comment:
We can extract the authorization logic into a private method to simplify the logic.